### DISCUSSION TITLE
How can I display MDX content correctly?

### DISCUSSION THREAD

**Discussion by gtvb** [chakra-ui/chakra-ui#2220] [Q&A] [answered]
I'm creating a blog using NextJS, MDX, and Chackra, and everything is working perfectly, except for the page where I render my markdown. The styles are not being displayed correctly - (what I mean by that is, h1, h2, h3...., a tags...) are not being showed the way they should. I know this is related to CSSReset, and I need it in my code to get everything working well, so, can someone give a suggestion on how to get through this situation? I really had to start a discussion here, because I posted a question at StackOverflow and nobody answered me :( 

If someone wants to check it, it's right here: [https://stackoverflow.com/questions/64288988/markdown-is-not-being-displayed-correctly-due-to-cssreset-chakra-ui](url)

I would be really grateful to get some feedbacks, I'm trying to solve this problem for a couple days already....

**Comment by ljosberinn** [comment, marked as answer]
from the looks of it, the issue isnt CSSReset, which you by the way have twice in there, since `ChakraProvider` renders it already, but your additional `EmotionThemeProvider`. `ChakraProvider` already renders one, then yours, so Chakra components will access your provider and try to retrieve their styling from there, which doesn't work. There's no need for this 2nd provider.

The Chakra docs use Next + mdx, so you can probably take some inspiration from it, most notably [next.config.js](https://github.com/chakra-ui/chakra-ui/blob/develop/website/next.config.js#L111) as well as [layouts/mdx](https://github.com/chakra-ui/chakra-ui/blob/develop/website/layouts/mdx.tsx).

**Reply by gtvb** [reply]
Can you take a look at my repo? Everything is there, and I think it will be easier to see what I'm missing there. Chakra docs is huge, and I don't know if I must use everything realted to the config, or from the layouts.

My app is still pretty small, and I don't have too many configurations there, so it will be really quick :)

If you have the time, it would really be great for me.

[https://github.com/Gabriel2233/gabriel-tiso.me](url)

**Reply by ljosberinn** [reply]
Yeah, this is most likely the issue here: https://github.com/Gabriel2233/gabriel-tiso.me/blob/master/contexts/ThemeContainer.tsx#L16

**Reply by gtvb** [reply]
Yeah I'm sorry, I haven't commited it yet, but I removed this provider and it didin't work.

**Reply by gtvb** [reply]
I got an answer at my StackOverflow post on this subject, is it the best way to do it??

Here it is: [https://stackoverflow.com/questions/64288988/markdown-is-not-being-displayed-correctly-due-to-cssreset-chakra-ui](url)

**Reply by gtvb** [reply]
That takes care of the problem, but if I want to style anything inside that page besides the markdown, for example, a header, or a footer, I won't be able to do it

**Reply by ljosberinn** [reply]
The comment is wrong because the docs over at https://next.chakra-ui.com are built with MDX.

Why would you not be able to style anything inside that page? [Here](https://github.com/chakra-ui/chakra-ui/blob/develop/website/src/components/mdx-components.tsx#L80) are plenty of custom MDX components.

**Reply by gtvb** [reply]
I saw that, and this components are used in a MDXProvider along with all chakraComponents. Do I have to wrap my context with this MDXProvider in order for things to work?

```
<MDXProvider components={{ ...chakraComponents, ...MDXComponents }}>
  //Some stuff
</MDXProvider>
```

One more thing, I tried to use this file (the one with the MDX Components) in my project, and I get an error due to importing chakra and useColorModeValue

```
import { Alert, Box, chakra, Kbd, useColorModeValue } from "@chakra-ui/core"

//Error

Module '"../node_modules/@chakra-ui/core/dist"' has no     
exported member 'chakra'.


Module '"../node_modules/@chakra-ui/core/dist"' has no     
exported member 'useColorModeValue'.
```

**Reply by ljosberinn** [reply]
No, MDXProvider is from a different MDX package, you're using the imo better one, we might migrate to it too.
In your case you should follow the example [here](https://github.com/hashicorp/next-mdx-remote#frontmatter--custom-processing). You seem to lack their setup entirely in the repo.

Your repo is still on v0.8, that's why. The docs are using the soon-to-be-released v1.
